src/main/java/com/visenze/productsearch/ProductSearch.java

```diff
@@ -0,0 +1,208 @@
+
+        /**
+         * Client configurations such as time-outs etc. These configurations
+         * will be used to construct the ViHttpClient class that is stored as
+         * member variable of ProductSearch
+         *
+         * @see ClientConfig
+         * @see ProductSearch
+         */
+        private ClientConfig config;
+
+        /**
+         * Constructor with only highly necessary parameters.
+         *
+         * @param app_key unique access key used as credentials
+         * @param placement_id which placement ID (from dashboard)
+         */
+        public Builder(String app_key, Integer placement_id) {
+            this.app_key      = app_key;
+            this.placement_id = placement_id;
+            this.config       = new ClientConfig();
+        }
+
+        /**
+         * Build the ProductSearch class based on configurations set through the
+         * builder pattern.
+         *
+         * @return ProductSearch object class
+         */
+        public ProductSearch build() {
+            return new ProductSearch(this.app_key, this.placement_id, endpoint, config);
+        }
+
+        /**
+         * Set api custom url endpoint for
+         *
+         * @param url The custom url endpoint to use instead
+         *
+         * @return this 'itself'
+         */
+        public Builder setApiEndPoint(String url) {
+            this.endpoint = url;
+            return this;
+        }
+
+        /**
+         * Set custom client configurations for http connections
+         *
+         * @param config custom client http configurations
+         *
+         * @return this 'itself'
+         *
+         * @see ClientConfig
+         */
+        public Builder setClientConfig(ClientConfig config) {
+            this.config = config;
+            return this;
+        }
+    }
+
+    /**
+     * Constructor of ProductSearch made private to only allow it's Builder
+     * class to create it.
+     *
+     * @param app_key Unique app key that acts as authenticator for the client
+     * @param placement_id Placement ID of the template that was chosen when
+     *                     creating apps on the dashboard
+     * @param endpoint Which endpoint to use for all ViHttpClient queries
+     * @param config Configuration to the behaviours of the ViHttpClient
+     */
+    private ProductSearch(String app_key, Integer placement_id, String endpoint, ClientConfig config) {
+        this.app_key      = app_key;
+        this.placement_id = placement_id;
+        this.endpoint     = endpoint;
+        this.httpClient   = new ViSearchHttpClientImpl(this.endpoint, app_key, placement_id.toString(), config);
+    }
+
+    /**
+     * Calls the POST method for the image search API
+     *
+     * @param params ImageSearchParam specific parameters to perform request
+     *
+     * @return ViSearchHttpResponse http response of search results
+     */
+    public ViSearchHttpResponse imageSearch(ImageSearchParam params) {
```

Comment:
need to create a ProductSearchHttpResponse instead of ViSearchHttpResponse, they have different format